### ISSUE TITLE
incorrect error warning about keyword arguments

### ISSUE THREAD

**Issue by wztdream** [microsoft/pyright#2021]
If we use keyword arguments, in some condition there will be an error warning:
`Argument expression after ** must be a mapping with a "str" key type`

Below  is an example, only B trigger this warning. I am not sure this is a bug or it is intended to work like this. If it is indented, it seems to me C should also trigger this warning, but it is not.
I can remove this warning by add a type check ` assert isinstance(kwargs, dict)`, but it seems to me pyright should not force user to do this type check, as python is type insensitive language and there will be an runtime error if the type is incorrect, so seems to me no need force user to check the type of kwargs.
``` python
class A():
    def __init__(self, name, **kwargs):
        "docstring"
        self.name = name
        self.kwargs = kwargs


class B():
    def __init__(self, name_list, kwargs_list):
        "docstring"
        self.As = []
        for name, kwargs in zip(name_list, kwargs_list):
            self.As.append(A(name, **kwargs))  # here will be error warning


class C():
    def __init__(self, name, kwargs):
        "docstring"
        self.A = A(name, **kwargs)


class D():
    def __init__(self, name, **kwargs):
        "docstring"
        self.A = A(name, **kwargs)

```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
If you want a workaround in the meantime, if you add a type annotation to one of the input parameters for the `__init__` method, the error will disappear. For example, if you add the type annotation `Sequence[str]` to the `name_list` parameter.

The bug is due to a bad interaction with a feature that we call "pseudo generic classes". When the constructor for a class is completely unannotated, pyright synthesizes types for the input parameters. Each parameter is given a generic type, as though you had manually added annotations `: T1`, `: T2`, etc. So even though `B` is not declared as a generic class, pyright internally treats it as such. This technique allows for better type inference within code that has no type annotations. In this particular example, the `name_list` and `kwargs_list` parameters were given these internally-synthesized types. When passed through the `zip` function, this resulted in a generic result that was then unpacked in the expression `**kwargs`, which triggered the bug.

**Comment by wztdream** [comment]
> If you want a workaround in the meantime, if you add a type annotation to one of the input parameters for the `__init__` method, the error will disappear. For example, if you add the type annotation `Sequence[str]` to the `name_list` parameter.
> 
> The bug is due to a bad interaction with a feature that we call "pseudo generic classes". When the constructor for a class is completely unannotated, pyright synthesizes types for the input parameters. Each parameter is given a generic type, as though you had manually added annotations `: T1`, `: T2`, etc. So even though `B` is not declared as a generic class, pyright internally treats it as such. This technique allows for better type inference within code that has no type annotations. In this particular example, the `name_list` and `kwargs_list` parameters were given these internally-synthesized types. When passed through the `zip` function, this resulted in a generic result that was then unpacked in the expression `**kwargs`, which triggered the bug.

That is great, thank you for quick reply and detail explain:)

**Comment by erictraut** [comment]
We'll leave the bug open until we release the next version.

**Comment by erictraut** [comment]
This is included in pyright 1.1.152, which I just published. It will also be included in the next release of pylance.
